### ISSUE TITLE
About thread interrupt

### ISSUE THREAD

**Issue by jburse** [leaningtech/cheerpj-appletrunner#18]
Dear Leaningtech,

Could already figure out how to work around this issue https://github.com/leaningtech/cheerpj-appletrunner/issues/14 but there is a residual issue. I now tried to verify that monitors are not interruptible. Pickeled my code as follows to show the problem:

Thread that gets repeatedly called and might get interrupted:

                while (ts.getBufferSize() == 0) {
                    try {
                        ts.wait(1000);
                        System.err.println("timeout or notify");
                    } catch (InterruptedException x) {
                        System.err.println("interrupt exception");
                        throw new InterruptedIOException();
                    }
                }

**Comment by jburse** [comment]
Here is what happens in the original JVM:

```
timeout or notify
timeout or notify
timeout or notify
interrupt exception
timeout or notify
timeout or notify
timeout or notify
```

**Comment by alexp-sssup** [comment]
Can you provide a fully self contained test case?

**Comment by jburse** [comment]
I see it in the applet, but this might not be a stripped down test case.
But in the applet it is seen as follows, I interrupt the reading of a query:

![image](https://user-images.githubusercontent.com/5411487/50080890-6bdc7e00-01ed-11e9-929c-6de538cb40c1.png)

And the reading of a query gets interrrupted (a new ?-, I know this looks a little strange, but this is only an example, the interrupt is also used for other stuff):

![image](https://user-images.githubusercontent.com/5411487/50080963-9595a500-01ed-11e9-98fc-122f084450d1.png)

**Comment by jburse** [comment]
With CheerpJ, since no InterruptedException is thrown, it simply does nothing,
after using the Run | Abort menu item:

![unbenannt](https://user-images.githubusercontent.com/5411487/50081158-10f75680-01ee-11e9-8452-22deb97a2654.PNG)

**Comment by jburse** [comment]
I could find a polling work-around, as follows:

                while (ts.getBufferSize() == 0) {
                    try {
                        if (AbstractLivestock.liveGetSignal(Thread.currentThread()) != null)
                            throw new InterruptedException();
                        ts.wait(1000);
                    } catch (InterruptedException x) {
                        throw new InterruptedIOException();
                    }
                }

Which works fine:

**Comment by alexp-sssup** [comment]
This problem should be fixed in CheerpJ Applet Runner 6.7
